### PR TITLE
emacs: Update mode file

### PR DESCRIPTION
**Pull request by HaoZeke**
This is a squashed commit including the following changes:
1) Update mode header
2) Clean up white-space
3) Fix free variable warning
4) Add proper file ending stuff
5) Rename to keep conventional naming scheme
6) Update the readme
7) Update to conform to `package-lint` criteria
8) Add license header

## Purpose

Basically this is done to conform to the norms allowing for `use-package` recipes.

## Author(s)

Rohit Goswami <r95g10@gmail.com>

## Backward Compatibility

No breaking changes

## Implementation Notes

The implementation was checked by [flycheck-cask](https://github.com/flycheck/flycheck-cask) and also by [package-lint](https://github.com/purcell/package-lint/).

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
N/A
- [ ] The source code follows the LAMMPS formatting guidelines
N/A

## Further Information, Files, and Links

N/A

### CODE REVIEW

File: tools/emacs/README.txt [hunk 1]
```diff
@@ -21,3 +22,14 @@ This file may also be called ~/.emacs.el, or ~/.emacs.d/init.el
...
 (1) Copy lammps.el to the directory ~/.emacs.d
 
+=Update:
+========
+
+The package may now also be installed by a QUELPA style recipe, namely:
+
+```lisp
+(use-package lammps-mode
+  :recipe (:fetcher url :url "https://github.com/lammps/lammps/blob/master/tools/emacs/lammps.el"))
```
Comment (akohlmey) [inline tools/emacs/README.txt]: Plese note, that it is generally not a good idea to refer to the master branch for software that people should be using for production. Best is likely to refer to the stable branch (we'll have a new stable release in a few weeks, so this update would be included there), or at least the unstable branch, which corresponds to the patch releases posted by Steve.

Comment (HaoZeke) [reply tools/emacs/README.txt]: Sure, actually I will update this with the MELPA repository recipe, so it's best not to merge yet...

**Comment by HaoZeke** [comment]
Should be ready as soon as https://github.com/melpa/melpa/pull/5659 is accepted.

**Comment by akohlmey** [comment]
One more suggestion. I just made a test with Xemacs, and this mode file doesn't seem to be compatible with it (with an old version, i get a hang, with a newer version an error). I would thus suggest to explicitly state, that this is a mode meant to be used for GNU Emacs and not XEmacs, even though the wording already implies that only GNU Emacs compatibility is intended.

**Comment by HaoZeke** [comment]
Could you direct me to a working font-locking file for Xemacs? [the AUR xemacs seems to be broken currently so I can't test directly].

Also, what was the error?

**Comment by akohlmey** [comment]
Here is the error message and stack trace from XEmacs  21.5  (beta34) "kale"

```
(5) (file-mode-spec/warning) Error in File mode specification: Symbol's function definition is void: shell-script-mode 
 
Backtrace follows: 
 
  backtrace(nil t) 
  # bind (standard-output standard-output handler-arg-name) 
  backtrace-in-condition-handler-eliminating-handler(#:call-trapping-errors-Ldc9FC5Hr) 
  # bind (errstr #:call-trapping-errors-Ldc9FC5Hr) 
  #<compiled-function (from "/builddir/build/BUILD/xemacs-beta/lisp/files.elc") (#:call-trapping-errors-Ldc9FC5Hr) "...(17)" [#:ca\ll-trapping-errors-Ldc9FC5Hr errstr error-message-string lwarn file-mode-spec warning "Error in %s: %s\n\nBacktrace follows:\n\n%s\" "File mode specification" backtrace-in-condition-handler-eliminating-handler] 8 0x6be>((void-function shell-script-mode)) 
  (shell-script-mode) 
```

**Comment by HaoZeke** [comment]
@akohlmey could you check if changing `shell-script-mode` to `sh-mode` fixes the error? I believe GNU emacs has the first aliased to the second..

**Comment by akohlmey** [comment]
@HaoZeke i played around with that. needed to add (require 'shell) to make that work. however, then xemacs gets stuck with processing the regular expressions. not really worth fixing IMO, since XEmacs seems to have fallen behind GNU emacs massively (it used to be the other way around).

**Comment by HaoZeke** [comment]
@akohlmey Thanks for letting me know, I'll drop it from the todo :smile:

**Pull request merged**